### PR TITLE
feat: adds testing to ai content generator

### PR DESCRIPTION
**Pull request by rusticpenguin**
## Purpose

We want to be able to confidently make changes to the AI Content Generator

## Approach

Testing components and hooks allows us to verify our changes do not break existing functionality.

### CODE REVIEW

File: apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        <GeneratedTextPanel
+          generate={() => {}}
+          ai={hook.result.current}
+          outputFieldValidation={null}
+          apply={() => {}}
+        />
+      </Tabs>
+    );
+
+    expect(getByText('4 characters')).toBeTruthy();
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx:46]: Where is this text `4 characters` coming from? It's not clear to me in the test if this is some default text.

File: apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    unmount();
+  });
+
+  it("Renders length error when output field doesn't meet length requirements", () => {
+    const hook = renderHook(() => useAI());
+    const { getByText, unmount, rerender } = render(
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx:52]: nit: `rerender` is unused in this test

File: apps/ai-content-generator/src/components/app/sidebar/DisplaySidebarWarning.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    expect(getByText('Chat GPT is currently unavailable.')).toBeTruthy();
+
+    rerender(<DisplaySidebarWarning hasBrandProfile={false} apiError={{ status: 503 }} />);
+
+    expect(getByText('Chat GPT is currently unavailable.')).toBeTruthy();
+    unmount();
+  });
+
+  it('Renders catch all error', () => {
+    const { getByText, rerender, unmount } = render(
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/sidebar/DisplaySidebarWarning.spec.tsx:51]: nit: `rerender` is unused in this test

File: apps/ai-content-generator/src/components/app/sidebar/DisplaySidebarWarning.tsx [hunk 4]
```diff
@@ -51,6 +51,7 @@ const DisplaySidebarWarning = (props: Props) => {
     }
   }
 
+  console.log(hasBrandProfile);
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/sidebar/DisplaySidebarWarning.tsx:54]: remove `console.log`

File: apps/ai-content-generator/src/components/app/sidebar/SidebarButtons.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const sdk = mockSdk.sdk;
+
+vi.mock('@contentful/react-apps-toolkit', () => ({
+  useSDK: () => sdk,
+  useCMA: () => mockCma,
+  useAutoResizer: () => {},
+}));
+
+describe('Sidebar Buttons', () => {
+  it('renders', () => {
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/sidebar/SidebarButtons.spec.tsx:17]: maybe worth adding a test for when `shouldDisableButtons` is `true`?

File: apps/ai-content-generator/src/components/app/sidebar/feature-button/FeatureButton.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { fireEvent, render } from '@testing-library/react';
```
Comment (whitelisab) [inline apps/ai-content-generator/src/components/app/sidebar/feature-button/FeatureButton.spec.tsx:1]: nit: `fireEvent` is unused in this file

File: apps/ai-content-generator/src/hooks/dialog/useDialogParameters.spec.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    mockSdk.reset();
+  });
+
+  const expectStateToBe = (result: DialogParameters, expected?: DialogInvocationParameters) => {
+    expect(result.isLoading).toBe(expected === undefined);
+    expect(result.feature).toBe(expected?.feature);
+    expect(result.entryId).toBe(expected?.entryId);
+  };
+
+  it('should load immediatly when parameters are present', () => {
```
Comment (whitelisab) [inline apps/ai-content-generator/src/hooks/dialog/useDialogParameters.spec.ts:29]: nit: `immediately`

File: apps/ai-content-generator/package.json [hunk 8]
```diff
@@ -54,13 +54,13 @@
     "@typescript-eslint/eslint-plugin": "^6.2.1",
     "@typescript-eslint/parser": "^6.2.1",
     "@vitejs/plugin-react": "4.0.4",
+    "@vitest/coverage-v8": "0.34.6",
```
Comment (whitelisab) [inline apps/ai-content-generator/package.json:57]: I like the addition of seeing coverage while running the tests!

**Review by whitelisab — APPROVED** [review]
This looks great to me! Mostly just some small comments, nothing blocking. Awesome work! ✨

File: apps/ai-content-generator/package.json [hunk 9]
```diff
@@ -25,7 +25,7 @@
     "start:staging": "NODE_ENV='staging' && npm run start:preview",
     "start:prod": "NODE_ENV='production' && npm run start:preview",
     "build": "tsc && vite build --mode ${NODE_ENV:='production'}",
-    "test": "vitest",
+    "test": "vitest --logHeapUsage --coverage",
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/package.json:28]: Yup! I want to separate those 2 prs, thank you for the callout!

File: apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx (hunk 1 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx:46]: The 4 chars are coming from line 33. where I'm setting the `{ output }` value from `useAI()` to "text"

File: apps/ai-content-generator/package.json (hunk 8 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/package.json:57]: Thanks! It's such a great tool. Seeing the % gives me joy

File: apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx (hunk 1 above)
Comment (whitelisab) [reply apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx:46]: Thank you! Totally missed that! 😅

**Pull request merged**